install/traf_checkset_env.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+#
+# make sure the environment can build the code of trafodion or not.
+# must configure the yum repo right before execute this script.
+# run this script with normal user, while must has sudo permission.
+
+#default path
+MY_JVM_PATH=${MY_JVM_PATH-"/usr/lib/jvm"}
+MY_JAVA_VER=${MY_JAVA_VER-"java-1.7.0-openjdk"}
+MY_SUDO=${MY_SUDO-"sudo"}
+MY_YUM=${MY_YUM-"${MY_SUDO} yum -y"}
+
+# for setup tools
+MY_LOCAL_SW_DIST=${MY_LOCAL_SW_DIST-${HOME}/local_software_tools}
+MY_INSTALL_SW_DIST=${MY_INSTALL_SW_DIST-${HOME}/installed_software_tools}
+MY_DOWNLOAD_SW_DIST=${MY_DOWNLOAD_SW_DIST-${HOME}/download_software_tools}
+
+CHECKLOG=${LOGFILE-$(pwd)/$0.log}
+local_sws=(udis llvm mpich bison icu zookeeper thrift apache-maven protobuf apache-log4cxx hadoop)
+http_sws=(
+http://sourceforge.net/projects/udis86/files/udis86/1.7/udis86-1.7.2.tar.gz
+http://llvm.org/releases/3.2/llvm-3.2.src.tar.gz
+http://www.mpich.org/static/downloads/3.0.4/mpich-3.0.4.tar.gz
+http://ftp.gnu.org/gnu/bison/bison-3.0.tar.gz
+http://download.icu-project.org/files/icu4c/4.4/icu4c-4_4-src.tgz
+https://archive.apache.org/dist/zookeeper/zookeeper-3.4.5/zookeeper-3.4.5.tar.gz
+http://archive.apache.org/dist/thrift/0.9.0/thrift-0.9.0.tar.gz
+http://archive.apache.org/dist/maven/maven-3/3.3.3/binaries/apache-maven-3.3.3-bin.tar.gz
+https://github.com/google/protobuf/releases/download/v2.5.0/protobuf-2.5.0.tar.gz
+https://dist.apache.org/repos/dist/release/logging/log4cxx/0.10.0/apache-log4cxx-0.10.0.tar.gz
+http://archive.apache.org/dist/hadoop/common/hadoop-2.6.0/hadoop-2.6.0.tar.gz
+)
+
+# check the local software directory
+if [ ! -d ${MY_LOCAL_SW_DIST} ]; then
+    echo "WARN: local sofrware tools aren't exist, should be downloading all
+    software tools from internet, it's very slowlly! While you can
+    prepare local software, and set environment MY_LOCAL_SW_DIST to
+    change default software path, and run this script again. The 
+    default local software directory is [${MY_LOCAL_SW_DIST}]. You
+    can set environment MY_LOCAL_SW_DIST to change it."
+    echo "Do you want continue? Enter y/n (default:n):"
+    read YN
+    case ${YN} in
+	Y|y|Yes|YES)
+	    ;;
+	*)
+	    echo "Download following build tools from internet:"
+	    for i in `seq ${#local_sws[@]}`
+	    do
+		printf "%2d.%15s: %s\n" ${i} ${local_sws[${i}-1]} ${http_sws[${i}-1]}
+	    done
+	    exit 1
+	    ;;
+    esac
+else
+    # check the local software's source exist or not
+    for local_sw in ${local_sws[@]}
+    do
+	local_file=`ls ${MY_LOCAL_SW_DIST} | grep ${local_sw}`
+	if [ "x${local_file}" = "x" ]; then
+            echo "WARN: [${local_sw}] source file not exist in directory [${MY_LOCAL_SW_DIST}]"
```

Comment:
Suggested wordsmith: "WARNING: [${local_sw}] source file does not exist in directory [${MY_LOCAL_SW_DIST}]"